ShinyExpenseTrackerTests/src/ca/ualberta/cs/shinyexpensetracker/test/AddExpenseItemTests.java

```diff
@@ -143,36 +143,23 @@ public void run() {
 				BigDecimal amount = new BigDecimal(10.00);
 				Bitmap bitmap = null;
 
-				ExpenseItem expense = new ExpenseItem("name", date,
-						Category.ACCOMODATION, amount, Currency.CAD,
+				ExpenseItem expense = new ExpenseItem("name", date, Category.ACCOMODATION, amount, Currency.CAD,
 						"description", bitmap);
 
 				assertEquals("name != name", "name", expense.getName());
-				assertNotSame("false positive, name", "Wrong Name",
-						expense.getName());
+				assertNotSame("false positive, name", "Wrong Name", expense.getName());
 				assertEquals("date != date", date, expense.getDate());
-				assertNotSame("false positive date", "wrong date",
-						expense.getDate());
-				assertEquals("category != accomodation", Category.ACCOMODATION,
-						expense.getCategory());
-				assertNotSame("false positive, category", "wrong category",
-						expense.getCategory());
-				assertEquals("amount != 10.00", amount,
-						expense.getAmountSpent());
-				assertNotSame("false positive, amount", new BigDecimal(5.00),
-						expense.getAmountSpent());
-				assertEquals("currnency != CAD", Currency.CAD,
-						expense.getCurrency());
-				assertNotSame("false positive, currency", "wrong currency",
-						expense.getCurrency());
-				assertEquals("description != description", "description",
-						expense.getDescription());
-				assertNotSame("false positibe description",
-						"wrong description", expense.getDescription());
-				assertEquals("bitmap != bitmap", bitmap,
-						expense.getReceiptPhoto());
-				assertNotSame("false posibive, photo", "not bitmap",
-						expense.getReceiptPhoto());
+				assertNotSame("false positive date", "wrong date", expense.getDate());
+				assertEquals("category != accomodation", Category.ACCOMODATION, expense.getCategory());
+				assertNotSame("false positive, category", "wrong category", expense.getCategory());
+				assertEquals("amount != 10.00", amount, expense.getAmountSpent());
+				assertNotSame("false positive, amount", new BigDecimal(5.00), expense.getAmountSpent());
+				assertEquals("currnency != CAD", Currency.CAD, expense.getCurrency());
+				assertNotSame("false positive, currency", "wrong currency", expense.getCurrency());
+				assertEquals("description != description", "description", expense.getDescription());
+				assertNotSame("false positibe description", "wrong description", expense.getDescription());
+				assertEquals("bitmap != bitmap", bitmap, expense.getReceiptPhoto());
+				assertNotSame("false posibive, photo", "not bitmap", expense.getReceiptPhoto());
```

Comment:
I know this isn't part of your issue, but can we scrap some of these `assertNotSame` checks? They're not really doing anything useful other than being a very weak sanity check. It was probably my fault that they're there, but `assertNotSame` is typically used for checking that an object is a clone or copy of an original (i.e. same values, different ids); we should avoid using it unless that's our intent and use `MoreAsserts.assertNotEqual( ... )` instead.

For instance, line 156 is definitely giving us a false positive: it'll pass even if you changed it to `assertNotSame(new BigDecimal(5.00), new BigDecimal(5.00))`.